packages/MapFactory/test/tstMapInterface.cpp

```diff
@@ -210,29 +211,84 @@ void test( bool &success, Teuchos::FancyOStream &out )
                          tgt_data.get() );
     TEST_EQUALITY( errno, DTK_SUCCESS );
 
-    // Create a map.
-    const std::string options = "{ \"Map Type\": \"Nearest Neighbor\" }";
     auto comm = Teuchos::getRawMpiComm( *teuchos_comm );
-    auto map_handle = DTK_createMap( SpaceSelector<MapSpace>::value(), comm,
-                                     src_handle, tgt_handle, options.c_str() );
-    TEST_EQUALITY( errno, DTK_SUCCESS );
 
-    // Apply the map.
-    DTK_applyMap( map_handle, "dummy", "dummy" );
-    TEST_EQUALITY( errno, DTK_SUCCESS );
+    // Check valid syntax in map create
+    for ( std::string const options : {
+              R"({ "Map Type": "Nearest Neighbor" })",
+              R"({ "Map Type": "NN" })",
+              R"({ "Map Type": "Moving Least Squares" })",
+              R"({ "Map Type": "MLS" })",
+              R"({ "Map Type": "MLS", "Order": "Linear" })",
+              R"({ "Map Type": "MLS", "Order": "1" })",
+              R"({ "Map Type": "MLS", "Order": "Quadratic" })",
+              R"({ "Map Type": "MLS", "Order": "2" })",
+          } )
+    {
+        auto map_handle =
+            DTK_createMap( SpaceSelector<MapSpace>::value(), comm, src_handle,
+                           tgt_handle, options.c_str() );
+        TEST_EQUALITY( errno, DTK_SUCCESS );
 
-    // Check the results. The points are the same and the current
-    // implementation of the map C interface uses nearest neighbor so we
-    // should get the same result.
-    for ( int p = 0; p < num_point; ++p )
+        DTK_destroyMap( map_handle );
+        TEST_EQUALITY( errno, DTK_SUCCESS );
+    }
+
+    // Check invalid syntax in map create
+    for (
+        std::string const options : {
+            R"({ "Map Type": "Nearest Neighbor " })", // trailing whitespace
+            R"({ "Map Type": "nearest neighbor" })",  // first letter not
```

Comment:
Ah OK - so `R` treats it as a raw string literal and you avoid having to use `\` before every `"`. Does this syntax also concatenate all of these key-value pairs into the options string?

Reply:
Nevermind - I see the loop structure now. You are testing to make sure all these things either work or don't work - nice!